### ISSUE TITLE
Multiple feature requests

### ISSUE THREAD

**Issue by ancientPrinter** [Mazahir26/koduko#4]
Hi, first of all, I wanna thank you for creating this great app. The UI is amazing and it works really smooth.

And sorry for the list, but I didn't want to bother you by creating separated issues for every requests.

1) I'd like to suggest a feature to add custom durations for single tasks: right now a task lasts, at max, 30 minutes and I think it's not enough sometimes. I know I could add multiple instances of the same task to increase the total time, but I think it would be a better idea to enter the duration manually.

2) Increase characters limit for tasks and routines names: I think 20 characters is quite short while trying to write a descriptive name for a task.

3) Add an option to reset/undo a routine: I marked a task as completed by accident and couldn't undo it, which messes up statistics.

4) Add a notification for running tasks: to see remaining time and current task/routine.

5) Don't start remaining tasks within a routine automatically: I started a task and left the app running in the background. It went through the rest of the tasks in that routine until there were no remaining tasks without alerts when one of them was finished.

Thanks in advance!

**Comment by Mazahir26** [comment]
First of all thanks for checking out the project!
You have a lot of feature request and i am glad that you found the app useful.
I am currently working on few of the features that you mentioned above notably a notification for currently running task.
But might end up adding all the features in the upcoming updates.

Thank you!

**Comment by Mazahir26** [comment]
Hey, With the new update I have added few of the suggested features.

1.  I strongly feel that a task should not have a duration of more than 30 mins (A person can't concentrate or continue working on a single task which has a duration of more than 25 mins without taking breaks). Moreover a user can just duplicate and can work on the same task for longer period.
2. I have nothing against long/descriptive names, its just that if the task or routine has a long name the UI design gets effected. I might add an field for optional descriptions in the future.
3. I have implemented Undo and Redo feature for a routine.
4. I was working on that feature, but the progress does not get updated when the app goes in background or screen gets locked which has no use.(There is a branch for it and I try to add that feature in the near future).
5.  Yes, Tasks will now not start automatically, and there will be a notification when the task gets completed.
